py/dml/structure.py

```diff
@@ -2592,6 +2592,11 @@ def mkmethod(site, rbrace_site, location, parent_obj, name, inp_ast,
             if t.is_int and t.is_endian:
                 raise EEARG(site)
 
+    for (n, t) in outp:
+        if t and deep_const(t):
+            raise ICE(site,
```

Comment:
ICEs that are known to be reachable should be accompanied with a jira-referencing comment. A failing test is good but not strictly necessary.